modules/cookbooks/pages/rag.adoc

```diff
@@ -0,0 +1,174 @@
+= Retrival Augmented Generation (RAG)
+// tag::single-source[]
+:description: How to configure Redpanda Connect to create a RAG pipeline using Postgres and PGVector.
+
+This cookbook demonstrates how to create a Retrival Augmented Generation (RAG) pipeline using Postgres and https://github.com/pgvector/pgvector[PGVector]. We'll create two Redpanda Connect pipelines: one for creating embeddings with https://ollama.ai[Ollama^] and another for searching the data using https://github.com/pgvector/pgvector[PGVector]. We will run both pipelines together using xref:guides/streams_mode/about.adoc[streams mode].
+
+== Compute the embeddings
+
+We'll start out by defining the indexing pipeline that will take textual data from Redpanda, compute embeddings for it, and then write it into a Postgres table with a PGVector index on the embeddings column.
+
+Articles to be indexed are received over the topic `articles` and look like this:
+
+[source,json]
+----
+{
+  "type": "article",
+  "article": {
+    "id": "123foo",
+    "title": "Dogs Stop Barking",
+    "content": "The world was shocked this morning to find that all dogs have stopped barking."
+  }
+}
+----
+
+We'll be able to read from our Redpanda topic using the xref:components:inputs/kafka.adoc[`kafka`] input:
+
+[source,yaml]
+----
+input:
+  kafka:
+    addresses: [ "${REDPANDA_CLUSTER}" ]
+    topics: [ articles ]
+    consumer_group: rp_connect_articles_group
+    tls:
+      enabled: true
+    sasl:
+      mechanism: SCRAM-SHA-256
+      user: "${REDPANDA_USER}"
+      password: "${REDPANDA_PASSWORD}"
+----
+
+We'll be using https://ollama.com/library/nomic-embed-text[Nomic Embed] to compute embeddings. Since each request only applies to a single document we will make this scale by making requests in parallel across our document batches.
```

Comment:
```suggestion
You can use https://ollama.com/library/nomic-embed-text[Nomic Embed] to compute embeddings. Since each request only applies to a single document, you can scale this by making requests in parallel across our document batches.
```